rmw_fastrtps_shared_cpp/src/rmw_publish.cpp

```diff
@@ -33,13 +33,14 @@ __rmw_publish(
   rmw_publisher_allocation_t * allocation)
 {
   (void) allocation;
-  RCUTILS_CHECK_FOR_NULL_WITH_MSG(publisher, "publisher pointer is null", return RMW_RET_ERROR);
   RCUTILS_CHECK_FOR_NULL_WITH_MSG(
-    ros_message, "ros_message pointer is null", return RMW_RET_ERROR);
+    publisher, "publisher pointer is null", return RMW_RET_INVALID_ARGUMENT);
+  RCUTILS_CHECK_FOR_NULL_WITH_MSG(
+    ros_message, "ros_message pointer is null", return RMW_RET_INVALID_ARGUMENT);
 
   if (publisher->implementation_identifier != identifier) {
     RMW_SET_ERROR_MSG("publisher handle not from this implementation");
-    return RMW_RET_ERROR;
+    return RMW_RET_INCORRECT_RMW_IMPLEMENTATION;
   }
```

Comment:
@Lobotuerk consider using `RMW_CHECK_TYPE_IDENTIFIERS_MATCH` instead, like:

```c++
  RMW_CHECK_TYPE_IDENTIFIERS_MATCH(
    pub,
    pub->implementation_identifier,
    identifier,
    return RMW_RET_INCORRECT_RMW_IMPLEMENTATION);
```

Reply:
```RMW_CHECK_TYPE_IDENTIFIERS_MATCH``` is not a part of https://github.com/ros2/rmw/blob/master/rmw/include/rmw/error_handling.h, where can I find it?

Reply:
Nvm, found it

Reply:
Changed it to use RMW standard macros